tools/print_tock_memory_usage.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/env python3
+
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Prints out the memory usage of a Tock kernel binary ELF.
+# Currently only works on ARM binaries.
+#
+# Usage: print_tock_memory_usage.py ELF
+#
+# Author: Philip Levis <philip.levis@gmail.com>
+
+# pylint: disable=superfluous-parens
+'''
+Script to print out the memory usage of a Tock kernel binary ELF.
+
+Usage: print_memory_usage.py ELF
+Options:
+  -dn, --depth=n      Group symbols at depth n or greater. E.g.,
+                      depth=2 will group all h1b::uart:: symbols
+                      together. Default: 1
+  -v, --verbose       Print verbose output.
+  -s, --show-waste    Show where RAM is wasted (due to padding)
+'''
+
+import os
+import re
+import sys
+import getopt
+import cxxfilt   # Demangling C++/Rust symbol names
+
+
+OBJDUMP = "llvm-objdump"
```

Comment:
perhaps use `rustup run llvm-objdump` as this will work as long as the user has installed the Rust LLVM tools through rustup. If they haven't, rustup should give a useful message for how to install. It also ensures that the version of `llvm-objdump` used corresponds to the version of the LLVM compiler `rustc` is using.

I don't feel strongly about this: I have `llvm-objdump` from my Linux distribution installed on my path

Reply:
picking hair... it is `rustup run nightly llvm-objdump`

Reply:
So what toolchain do I specify, then? This starts to get tricky.

Reply:
Given it's `llvm-objdump`, `stable` seems like the right one to use?

Reply:
Oy... right... never mind then, it's not better... Let's stick to `llvm-objdump` as is.